File: crates/zymatica-language-u/01_Language_U_Taxonomy/src/java/Proof.java
// Watermark: ip zymatica.space | astronautshe.com
// Copyright (c) 2026 Zymatica. All rights reserved.

public class Proof {
    public static void main(String[] args) {
        System.out.println("======================================================================");
        System.out.println("ZYMATICA | Language-U Taxonomy Proof (Java Edition)");
        System.out.println("======================================================================\n");

        String[] messages = {
            "SYSTEM_ALERT: SX1302 reset line high, restarting gateway transceiver.",
            "GATEWAY_STATUS: Temperature 42C, LoRa SNR 9.2dB, packets active.",
            "COMMAND_ROUTE: Directing node 04 to lower power state (TxPower 14dBm)."
        };
        int totalRawBits = 0;
        for (String m : messages) {
            totalRawBits += m.length() * 8;
        }
        int totalSemanticBits = messages.length * 24;
        double savings = (1.0 - ((double)totalSemanticBits / totalRawBits)) * 100.0;
        System.out.println("[1] Total Raw bits: " + totalRawBits);
        System.out.println("[2] Total Semantic bits: " + totalSemanticBits);
        System.out.printf("[3] Space savings: %.2f%%\n", savings);

        System.out.println("\n[VERIFICATION] Semantic decomposition limits proven. Task-Oriented Semantic Rate-Distortion Verified.");
    }
}

File: crates/zymatica-language-u/01_Language_U_Taxonomy/src/kotlin/proof.kt
// Watermark: ip zymatica.space | astronautshe.com
// Copyright (c) 2026 Zymatica. All rights reserved.

import java.io.File

fun main() {
    println("======================================================================")
    println("ZYMATICA | Language-U Taxonomy Proof (Kotlin Edition)")
    println("======================================================================\n")
    val messages = arrayOf(
        "SYSTEM_ALERT: SX1302 reset line high, restarting gateway transceiver.",
        "GATEWAY_STATUS: Temperature 42C, LoRa SNR 9.2dB, packets active.",
        "COMMAND_ROUTE: Directing node 04 to lower power state (TxPower 14dBm)."
    )
    var totalRawBits = 0
    for (m in messages) {
        totalRawBits += m.length * 8
    }
    val totalSemanticBits = messages.size * 24
    val savings = (1.0 - (totalSemanticBits.toDouble() / totalRawBits)) * 100.0
    println("[1] Total raw bits: $totalRawBits")
    println("[2] Total semantic bits: $totalSemanticBits")
    println(String.format("[3] Space savings: %.2f%%", savings))
    println("\n[VERIFICATION] Semantic decomposition limits proven. Task-Oriented Semantic Rate-Distortion Verified.")
}
